nautobot_device_lifecycle_mgmt/jobs/cve_tracking.py

```diff
@@ -65,3 +70,217 @@ def run(self, data, commit):  # pylint: disable=too-many-locals
 
         diff = VulnerabilityLCM.objects.count() - count_before
         self.log_success(message=f"Processed {cves.count()} CVEs and generated {diff} Vulnerabilities.")
+
+
+class NistCveSyncSoftware(Job):
+    """Checks all software in the DLC Plugin for NIST recorded vulnerabilities."""
+
+    name = "Find current NIST CVE for Software in Database"
+    description = """Searches the NIST DBs for CVEs related to software"""
+    read_only = False
+
+    class Meta:  # pylint: disable=too-few-public-methods
+        """Meta class for the job."""
+
+        commit_default = True
+        hidden = True
+
+    def run(self, data, commit):
+        """Check all software in DLC against NIST database and associate registered CVEs.  Update when necessary."""
+        all_software = SoftwareLCM.objects.all()
+        cve_counter = 0
+
+        for software in all_software:
+            manufacturer = str(software.device_platform.manufacturer).lower()
+            platform = self.validate_platform_name(str(software.device_platform.name).lower())
+            version = self.validate_platform_version(str(software.version))
+
+            cpe_software_search_url = self.create_cpe_software_search_url(manufacturer, platform, version)
+            software_cve_info = self.get_cve_info(cpe_software_search_url, software.id)
+
+            cve_counter += len(software_cve_info)
+            self.create_dlc_cves(software.id, software_cve_info)
+
+        self.log_success(
+            message=f"""Performed discovery on all software meeting
+                        naming standards.  Added {cve_counter} CVE."""
+        )
+        self.update_cves()
+
+    def validate_platform_name(self, platform: str) -> str:
+        """Platform names are typically entered with a preceding manufacturername.
+
+        Remove the preceding Manufacturer name and return only a platform
+        by splitting on the first space and replace spaces with underscores to
+        prepare.
+        """
+        platform = platform.split(" ", 1)[1].lower()
+        platform = platform.replace(" ", "_")
+
+        return platform
+
+    def validate_platform_version(self, version: str) -> str:
+        """Platform versions may contain characters that need to be escaped.
+
+        Create a compatible version.
+        If any additional chars need to be escaped, add to the list.
+        """
+        escape_list = [r"\(", r"\)"]
+        for escape_char in escape_list:
+            if re.search(escape_char, version):
+                version = re.sub(escape_char, "\\" + escape_char, version)
+
+        return version
+
+    def create_cpe_software_search_url(self, manufacturer: str, platform: str, version: str) -> str:
+        """Return the url for a cpe search against the NIST DB."""
+        base_url = 'f"""https://services.nvd.nist.gov/rest/json/cpes/1.0?addOns=cves&cpeMatchString=cpe:2.3:*:"""'
```

Comment:
This doesn't seem right 
```
>>> base_url = 'f"""https://services.nvd.nist.gov/rest/json/cpes/1.0?addOns=cves&cpeMatchString=cpe:2.3:*:"""'
>>> print(base_url)
f"""https://services.nvd.nist.gov/rest/json/cpes/1.0?addOns=cves&cpeMatchString=cpe:2.3:*:"""
>>>
```

Reply:
You are correct.  Blind following of flake8 recommendation for missing placeholders.  Pushed a commit with fix.